spec/std/range_spec.cr

```diff
@@ -79,6 +79,89 @@ describe "Range" do
     (1...5).includes?(5).should be_false
   end
 
+  it "does union" do
+    # Int ranges
+    (1..5).union(3..7).should eq(1..7)    # Overlapping integer ranges
+    (1..5).union(6..10).should eq(1..10)  # Adjacent ranges
+    (1..5).union(10..15).should eq(1...1) # Disjoint integer ranges
```

Comment:
I believe this relays the intention better:

```suggestion
    (1..5).union(10..15).should be_empty # Disjoint integer ranges
```

ditto elsewhere

Reply:
Using `be_empty` here results in an issue with one of the range macros:

```
Showing last frame. Use --error-trace for full trace.

There was a problem expanding macro 'macro_4849671408'

Code in src/range.cr:120:5

 120 | {% if E == Nil %}
       ^
Called macro defined in src/range.cr:120:5

 120 | {% if E == Nil %}

Which expanded to:

 > 3 | while end_value.nil? || current < end_value
 > 4 |   yield current
 > 5 |   current = current.succ
                           ^---
Error: undefined method 'succ' for Float64
```
`.empty?` must leverage this macro? I'm not sure, but I also don't think we necessarily need to use `be_empty` to in the spec.